editoast/src/views/layers/mod.rs

```diff
@@ -80,10 +80,21 @@ async fn layer_view(
         return Err(LayersError::new_view_not_found(view_slug, layer).into());
     }
 
-    let tiles_url_pattern = format!(
-        "{root_url}/layers/tile/{layer_slug}/{view_slug}/{{z}}/{{x}}/{{y}}/?infra={infra}",
-        root_url = get_root_url()
-    );
+    let mut tiles_url_pattern = get_root_url()?;
+    tiles_url_pattern
+        .set_path(format!("layers/tile/{layer_slug}/{view_slug}/{{z}}/{{x}}/{{y}}/").as_str());
+    tiles_url_pattern
+        .query_pairs_mut()
+        .append_pair("infra", &infra.to_string());
+
+    // this function will replace special characters like %7B with {
+    let tiles_url_pattern: url::form_urlencoded::Parse<'_> =
+        url::form_urlencoded::parse(tiles_url_pattern.as_str().as_bytes());
+    let tiles_url_pattern = tiles_url_pattern
+        .into_iter()
+        .map(|(key, value)| format!("{}={}", key, value))
+        .collect::<Vec<String>>()
+        .join("&");
```

Comment:
I prefer to not use `URL` in this case. We are not really crafting a URL but a URL template.